### PR TITLE
test(autoware_tensorrt_plugins): add SegmentCSR kernel tests

### PR DESCRIPTION
**Pull request by mojomex**
## Summary

First PR in the SegmentCSR split stack.

Adds parameterized reference-kernel coverage for SegmentCSR mean and max reductions using the existing launcher API. The covered cases include normal non-empty segments, empty middle/boundary segments, no-segment output, empty source input, empty-source/empty-segment output, and zero-column output.

This also adds the minimal launcher guard needed for those edge cases: zero-output work now returns before zero-size kernel launches, and output size is computed without dividing by an empty reduction axis.

## Stack

1. This PR: add edge-case unit tests with minimal launcher guard/wiring.
2. Follow-up: document the plugin contract and clean up in/out API names.
3. Follow-up: remove the dead n-dimensional `indptr` path.
4. Follow-up: keep SegmentCSR allocation-free (#12555).

### CODE REVIEW

File: perception/autoware_tensorrt_plugins/test/segment_csr_kernel_test.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+TEST(ReferenceKernelsTest, SegmentCsrMaxMatchesCpuReference)
+{
+  SKIP_TEST_IF_CUDA_UNAVAILABLE();
+
+  const std::size_t rows = 6U;
+  const std::size_t cols = 2U;
+  const std::vector<float> src{1.0F, 6.0F, 3.0F, 4.0F,  5.0F,  9.0F,
+                               7.0F, 8.0F, 2.0F, 10.0F, 11.0F, 12.0F};
+  const std::vector<std::int64_t> indptr{0, 2, 5, 6};
```
[the remark this answers is not in the document]
Comment (mojomex) [reply perception/autoware_tensorrt_plugins/test/segment_csr_kernel_test.cpp]: Addressed in 80003f4

File: perception/autoware_tensorrt_plugins/test/segment_csr_kernel_test.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+TEST(ReferenceKernelsTest, SegmentCsrMeanMatchesCpuReference)
+{
+  SKIP_TEST_IF_CUDA_UNAVAILABLE();
+
+  const std::size_t rows = 6U;
+  const std::size_t cols = 2U;
+  const std::vector<float> src{1.0F, 2.0F, 3.0F, 4.0F,  5.0F,  6.0F,
+                               7.0F, 8.0F, 9.0F, 10.0F, 11.0F, 12.0F};
+  const std::vector<std::int64_t> indptr{0, 2, 5, 6};
```
[the remark this answers is not in the document]
Comment (mojomex) [reply perception/autoware_tensorrt_plugins/test/segment_csr_kernel_test.cpp]: Addressed in 80003f4

**Review by manato — APPROVED** [review]
@mojomex 
LGTM! Thank you for your contribution!

**Comment by mojomex** [comment]
@amadeuszsz @ktro2828 @MasatoSaeki I'd appreciate if one of you code owners could review this :pray:

**Review by ktro2828 — APPROVED** [review]
LGTM!
